api_v3/lib/KalturaLiveEntryService.php

```diff
@@ -204,49 +204,52 @@ private function ingestAsset(entry $entry, $dbAsset, $filename)
-	 * @param KalturaMediaServerIndex $mediaServerIndex Media server index primary / secondary
+	 * @param KalturaEntryServerNodeType $mediaServerIndex Media server index primary / secondary
 	 * @param string $applicationName the application to which entry is being broadcast
-	 * @param KalturaLiveEntryStatus $liveEntryStatus the new status KalturaLiveEntryStatus::PLAYABLE | KalturaLiveEntryStatus::BROADCASTING
+	 * @param KalturaEntryServerNodeStatus $liveEntryStatus the new status KalturaLiveEntryStatus::PLAYABLE | KalturaLiveEntryStatus::BROADCASTING
 	 * @return KalturaLiveEntry The updated live entry
 	 *
 	 * @throws KalturaErrors::ENTRY_ID_NOT_FOUND
-	 * @throws KalturaErrors::MEDIA_SERVER_NOT_FOUND
+	 * @throws KalturaErrors::SERVER_NODE_NOT_FOUND
+	 * @throws KalturaErrors::ENTRY_SERVER_NODE_MULTI_RESULT
 	 */
-	function registerMediaServerAction($entryId, $hostname, $mediaServerIndex, $applicationName = null, $liveEntryStatus = KalturaLiveEntryStatus::PLAYABLE)
+	function registerMediaServerAction($entryId, $hostname, $mediaServerIndex, $applicationName = null, $liveEntryStatus = KalturaEntryServerNodeStatus::PLAYABLE)
 	{
-		$this->dumpApiRequest($entryId);
 		KalturaLog::debug("Entry [$entryId] from mediaServerIndex [$mediaServerIndex] with liveEntryStatus [$liveEntryStatus]");
 
 		$dbEntry = entryPeer::retrieveByPK($entryId);
 		if (!$dbEntry || !($dbEntry instanceof LiveEntry))
 			throw new KalturaAPIException(KalturaErrors::ENTRY_ID_NOT_FOUND, $entryId);
 
-		try {
-			if ($liveEntryStatus == KalturaLiveEntryStatus::BROADCASTING){
-				$dbEntry->setLiveStatus(KalturaLiveEntryStatus::BROADCASTING, $mediaServerIndex);
-			}
-			else {
-				$dbEntry->setMediaServer($mediaServerIndex, $hostname, $applicationName);
-			}
-		}
-		catch(kCoreException $ex)
+		$dbEntryServerNode = EntryServerNodePeer::retrieveByEntryIdAndServerType($entryId, $mediaServerIndex);
+
+		$dbServerNode = ServerNodePeer::retrieveActiveMediaServerNode($hostname);
+		if (!$dbServerNode)
+			throw new KalturaAPIException(KalturaErrors::SERVER_NODE_NOT_FOUND, $hostname);
+
+		if (!$dbEntryServerNode)
 		{
-			$code = $ex->getCode();
-			switch($code)
-			{
-				case kCoreException::MEDIA_SERVER_NOT_FOUND :
-					throw new KalturaAPIException(KalturaErrors::MEDIA_SERVER_NOT_FOUND, $hostname);
-				default:
-					throw $ex;
-			}
+			$liveEntryServerNode = new KalturaLiveEntryServerNode();
+			$dbLiveEntryServerNode = $liveEntryServerNode->toInsertableObject();
+			$dbLiveEntryServerNode->setEntryId($entryId);
+			$dbLiveEntryServerNode->setServerType($mediaServerIndex);
+			$dbLiveEntryServerNode->setPartnerId($this->getPartnerId());
+		} else {
+			$dbLiveEntryServerNode = $dbEntryServerNode;
 		}
 
+		/* @var $dbLiveEntryServerNode LiveEntryServerNode */
+		$dbLiveEntryServerNode->setServerNodeId($dbServerNode->getId());
+		$dbLiveEntryServerNode->setStatus($liveEntryStatus);
+		$dbLiveEntryServerNode->setUpdatedAt(time());
```

Comment:
Previously we never updated the kMediaServer time attribute ... why do we need to update for each register the updateAt value now?